src/main/java/hudson/tasks/test/AggregatedTestResultPublisher.java

```diff
@@ -126,13 +127,9 @@ public static final class TestResultAction extends AbstractTestResultAction {
         private final boolean includeFailedBuilds;
 
         /**
-         * The last time the fields of this object is computed from the rest.
+         * flag set when we need to recompute the aggregation.
          */
-        private transient long lastUpdated = 0;
-        /**
-         * When was the last time any build completed?
-         */
-        private static long lastChanged = 0;
+        private static final AtomicBoolean RECOMPUTE_NEEDED = new AtomicBoolean(true);
```

Comment:
I don't think this works correctly as there will be an action in multiple projects.  and all of them need to recompute, not just the first one.